### PR TITLE
Feature: Auto linking of accounts

### LINKED ISSUE #44: Auto Linking of accounts
Thank you for the plugin!

Our current scenario is the usage of the plugin in following context:
 - keycloak as OpenID connect provider and SSO Provider
 - manually creating accounts (random passwords) in Matomo by Matomo admins, users should not be created by the LoginOIDC plugin.
 - not every SSO user should have access to Matomo
 - Since keycloak is used as SSO Provider, we don't have user passwords in Matomo. So users can not login to matomo directly and linking of OIDC accounts can not be done by the users themselves.
 - We want to match users by their username (means: matomo user/login ID = keycloak user/login ID)

So our idea was to request a new feature called "auto linking of accounts".

### PR DESCRIPTION
**Pull request by ghost**
fixes #44

### CODE REVIEW

**Comment by dlecan** [comment]
This feature is very interesting!
1. I can provide French translations
2. Is it possible to customize the OIDC field used to match internal/external user?
2. is it possible to disable password authentication at all?

**Comment by ghost** [comment]
@dlecan

ad 1: yes please
ad 2: I hope I understood it correct, but I guess you can change the "userinfoId"-setting (https://github.com/dominik-th/matomo-plugin-LoginOIDC/blob/4.x-dev/SystemSettings.php#L89)
ad 3: I currently found no way to do that.

**Comment by pschichtel** [comment]
@dominik-th anything blocking this?

**Comment by Bengrunt** [comment]
That would be great to have this feature :)

**Comment by ghost** [comment]
Is there anything I can do to help with reviewing / evaluating this PR?

**Comment by dominik-th** [comment]
Hi!

Thank you for your PR!

Sorry about the delay, I am going to review and merge it this weekend!

File: SystemSettings.php [hunk 1]
```diff
@@ -46,6 +46,11 @@ class SystemSettings extends \Piwik\Settings\Plugin\SystemSettings
      */
     public $bypassTwoFa;
 
+    /**
+     * Enable auto linking of accounts
+     */
```
Comment (dominik-th) [inline SystemSettings.php:53]: Please add the datatype of the variable in here

File: Controller.php [hunk 2]
```diff
@@ -259,6 +259,22 @@ public function callback()
...
+        	$matomoUser = $userModel->getUser($providerUserId);
+		if (! empty($matomoUser)) {
+			if (empty($user)) {
+				$this->linkAccount($providerUserId, $providerUserId);
+			}
+	
+        		$user = $this->getUserByRemoteId("oidc", $providerUserId);
+		}
+	}
+	//
```
Comment (dominik-th) [inline Controller.php]: Formatting here is a little bit off.
Please use 4 space indentation as in the rest of the code.

Please add one or two inline comments describing what is happening

**Review by dominik-th — CHANGES_REQUESTED** [review]
Thank you for the PR!

I noticed a few minor things I'd like you to change before merging it.

**Comment by ghost** [comment]
I put the changes in one commit and hope that I haven't missed anything?

**Comment by ghost** [comment]
Have I missed something?

**Comment by dominik-th** [comment]
No, looks good! I'm going to test merge and release a new version next weekend

**Pull request merged**